Userland/Libraries/LibJS/Runtime/FinalizationRegistry.cpp

```diff
@@ -58,18 +58,16 @@ void FinalizationRegistry::remove_dead_cells(Badge<Heap>)
 }
 
 // 9.13 CleanupFinalizationRegistry ( finalizationRegistry ), https://tc39.es/ecma262/#sec-cleanup-finalization-registry
-void FinalizationRegistry::cleanup(FunctionObject* callback)
+ThrowCompletionOr<void> FinalizationRegistry::cleanup(FunctionObject* callback)
 {
-    auto& vm = this->vm();
     auto cleanup_callback = callback ?: m_cleanup_callback;
     for (auto it = m_records.begin(); it != m_records.end(); ++it) {
         if (it->target != nullptr)
             continue;
-        (void)call(global_object(), *cleanup_callback, js_undefined(), it->held_value);
+        (void)TRY(call(global_object(), *cleanup_callback, js_undefined(), it->held_value));
```

Comment:
This is not functionally equivalent. The old way would only return after removing, regardless of the exception state.

Spec:

```
a. Choose any such cell.
b. Remove cell from finalizationRegistry.[[Cells]].
c. Perform ? [HostCallJobCallback](https://tc39.es/ecma262/#sec-hostcalljobcallback)(callback, undefined, « cell.[[HeldValue]] »).
```

Reply:
Fixed and added comments so it's easier to check.